sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -317,12 +341,51 @@ object DottyPlugin extends AutoPlugin {
 
       // Because managedScalaInstance is false, sbt won't add the standard library to our dependencies for us
       libraryDependencies ++= {
-        if (isDotty.value && autoScalaLibrary.value)
-          Seq(scalaOrganization.value %% "dotty-library" % scalaVersion.value)
-        else
+        if (isDotty.value && autoScalaLibrary.value) {
+          val name =
+            if (isDottyJS.value) "dotty-library_sjs1"
+            else "dotty-library"
+          Seq(scalaOrganization.value %% name % scalaVersion.value)
+        } else
           Seq()
       },
 
+      // Patch up some more options if this is Dotty with Scala.js
+      scalacOptions := {
+        val prev = scalacOptions.value
+        /* The `&& !prev.contains("-scalajs")` is future-proof, for when sbt-scalajs adds that
+         * option itself but sbt-dotty is still required for the other Dotty-related stuff.
+         */
+        if (isDottyJS.value && !prev.contains("-scalajs")) prev :+ "-scalajs"
+        else prev
+      },
+      libraryDependencies := {
+        val prev = libraryDependencies.value
+        if (!isDottyJS.value) {
+          prev
+        } else {
+          prev
+            /* Remove the dependencies we don't want:
+             * * We don't want scalajs-library, because we need the one that comes
+             *   as a dependency of dotty-library_sjs1
+             * * We don't want scalajs-compiler, because that's a compiler plugin,
+             *   which is replaced by the `-scalajs` flag in dotc.
+             */
+            .filterNot { moduleID =>
+              moduleID.organization == "org.scala-js" && (
+                moduleID.name == "scalajs-library" || moduleID.name == "scalajs-compiler"
+              )
+            }
+            // Apply withDottyCompat to the dependency on scalajs-test-bridge
```

Comment:
I assume this applies transitively to `scalajs-test-interface`?

Reply:
Yes, because the dependency on scalajs-test-interface will be lookup up in its POM file. And in the POM file, this has already been hard-wired to scalajs-test-interface_2.13.